GeomIO/SharedIO.F90

```diff
@@ -13,9 +14,128 @@ module mapl3g_SharedIO
    public get_mapl_geom
    public create_time_variable
    public bundle_to_metadata
+   public esmf_to_pfio_type
+   public create_local_start
+   public create_global_count
+   public create_global_start
+   public create_file_shape
 
    contains
 
+   function create_file_shape(grid, field_shape, rc) result(file_shape)
+      integer, allocatable :: file_shape(:)
+      type(ESMF_Grid), intent(in) :: grid
+      integer, intent(in) :: field_shape(:)
+      integer, intent(out), optional :: rc
+
+      integer :: status, sz, ungr, tile_count
+      call ESMF_GridGet(grid, tileCount=tile_count, _RC)
+      sz = size(field_shape)
```

Comment:
If find `sz` to be a very confusing name here.   We usually use that for the size of arrays, but the size of the _shape_ of an array is its _rank_.     Perhaps you could see your way to replacing it with `n_dims` or something like that?